cdk/src/open-data-platform/data-plane/data-import/write-violations-data.handler.ts

```diff
@@ -0,0 +1,260 @@
+                                                 pws_id,
+                                                 violation_code,
+                                                 compliance_status,
+                                                 start_date, geom)
+                      VALUES ${sql.join(
+                        rows.map((row: ViolationsTableRow) => {
+                          return sql`(${row.violation_id}, 
+                                      ${row.pws_id}, 
+                                      ${row.violation_code}, 
+                                      ${row.compliance_status}, 
+                                      ${row.start_date},
+                                      ${sql.__dangerous__rawValue(
+                                        `ST_AsText(ST_GeomFromGeoJSON('${row.geom}'))`,
+                                      )})`;
+                        }),
+                        ',',
+                      )};`);
+}
+
+/**
+ * Inserts all rows into the violations table.
+ */
+async function deleteRows(db: ConnectionPool): Promise<any[]> {
+  return db.query(sql`DELETE
+                      FROM epa_violations
+                      WHERE violation_id IS NOT NULL`);
+}
+
+/**
+ * Reads the S3 CSV file and the number of rows successfully written.
+ */
+function parseS3IntoViolationsTableRow(
+  s3Params: AWS.S3.GetObjectRequest,
+  db: ConnectionPool,
+  numberOfRowsToWrite = DEFAULT_NUMBER_ROWS_TO_INSERT,
+): Promise<number> {
+  return new Promise(function (resolve, reject) {
+    const batchSize = 10;
+    let numberRows = 0;
+    let results: ViolationsTableRow[] = [];
+
+    const fileStream = S3.getObject(s3Params).createReadStream();
+    let pipeline = chain([
+      fileStream,
+      Pick.withParser({ filter: 'features' }),
+      streamArray(),
+      new Batch({ batchSize: batchSize }),
+    ]);
+
+    pipeline
+      .on('data', async (batch: any[]) => {
+        if (numberRows < numberOfRowsToWrite) {
+          for (const data of batch) {
+            const value = data.value;
+            const properties = value.properties;
+
+            console.log(properties);
+            if (LEAD_AND_COPPER_VIOLATIONS.has(properties[VIOLATION_CODE])) {
+              const startDate = moment(properties[COMPL_PER_BEGIN_DATE], 'DD-MMM-YY');
+              const row = new ViolationsTableRowBuilder()
+                .violationId(properties[VIOLATION_ID])
+                .pwsId(properties[PWSID])
+                .violationCode(properties[VIOLATION_CODE])
+                .complianceStatus(
+                  COMPLIANCE_STATUS_MAP.get(properties[COMPLIANCE_STATUS_CODE]) ?? '',
+                )
+                .startDate(startDate.format('YYYY-MM-DD'))
+                .geom(JSON.stringify(value.geometry))
+                .build();
+              results.push(row);
+
+              // Every batch size, write into the db.
+              if (results.length == batchSize) {
```

Comment:
nit: why not just put after the for loop instead of checking that results length == batch length?

Reply:
Renamed variables to make this clearer. The batch is not a batch. It's just a row